### DISCUSSION TITLE
Difference between neural receiver implemented in Part 4: Toward Learned Receivers and Neural Receiver for OFDM SIMO Systems ?

### DISCUSSION THREAD

**Discussion by omsahoowork** [NVlabs/sionna#314] [Q&A]
I tried implementing both the neural receivers and I found that Neural Receiver implemented in Neural Receiver for OFDM SIMO Systems performs better. I am not able to figure out the exact difference between the two. I have tried implementing it for all channel models including tdl, cdl, umi, uma and rma.
I only made a observation that both of them use different loss function calculations. But it seems they are similar in theoretical implementations. 
So can anyone help me with what exactly is the second one doing better ?

And i had been going through the references and I am curious that will the performance become more better if we increase the number of rest blocks used inside the neural receiver implementation ?

**Comment by jhoydis** [comment]
Could you be more specific with respect to which neural receivers you are referring to. In this [tutorial notebook](https://nvlabs.github.io/sionna/examples/Neural_Receiver.html#Neural-Receiver) there is only a single model trained.

**Reply by omsahoowork** [reply]
I am referring to the one implemented here  [link](https://nvlabs.github.io/sionna/examples/Sionna_tutorial_part4.html)

**Reply by jhoydis** [reply]
So, you are wondering why and how the neural receivers in [this](https://nvlabs.github.io/sionna/examples/Neural_Receiver.html#Neural-Receiver) and [this](https://nvlabs.github.io/sionna/examples/Sionna_tutorial_part4.html) notebook are different?

Maybe @faycalaa can answer to this.

**Reply by faycalaa** [reply]
Hi @OmSahoo-Work,

The neural receivers in these two notebooks share the same architecture and are both trained on the binary cross-entropy.

**Reply by omsahoowork** [reply]
Hi @faycalaa,  but I have implemented both on all the 5 3gpp channels and found that [this](https://nvlabs.github.io/sionna/examples/Neural_Receiver.html#Neural-Receiver) performs better on all channel conditions.
Also different loss functions are implemented in the code, so I was confused how this loss function difference makes such a large difference in performance. 
The one implemented on part 4 tutorial gives poor performance.

**Reply by faycalaa** [reply]
Hi @OmSahoo-Work ,

Both notebooks optimize the neural network-based receiver by minimizing the binary cross-entropy (BCE).

This is done directly in the [Part 4 notebook](https://nvlabs.github.io/sionna/examples/Sionna_tutorial_part4.html) by using the BCE as loss function.

In the ["Neural Receiver for OFDM SIMO Systems" notebook](https://nvlabs.github.io/sionna/examples/Neural_Receiver.html#Neural-Receiver), the additive inverse of the BMD rate is used as loss function, which is equivalent to minimizing the BCE. This can be seen from the definition of the BMD rate provided at the beginning of the section "End-to-end system", or from the way it is computed in cell 11.

**Reply by omsahoowork** [reply]
Hi @faycalaa, I wonder how come the performance is so different when I tested the same code for UMA, UMI and RMA channel conditions, without doing any changes in the model ?
